pkg/setting/setting_unified_alerting.go

```diff
@@ -95,47 +95,50 @@ func (u *UnifiedAlertingSettings) IsEnabled() bool {
 	return u.Enabled == nil || *u.Enabled
 }
 
+// readUnifiedAlertingEnabledSettings reads the settings for unified alerting.
+// It returns a non-nil bool and a nil error when unified alerting is enabled either
+// because it has been enabled in the settings or by default. It returns nil and
+// a non-nil error both unified alerting and legacy alerting are enabled at the same time.
 func (cfg *Cfg) readUnifiedAlertingEnabledSetting(section *ini.Section) (*bool, error) {
-	enabled, err := section.Key("enabled").Bool()
-	// the unified alerting is not enabled by default. First, check the feature flag
+	// At present an invalid value is considered the same as no value. This means that a
+	// spelling mistake in the string "false" could enable unified alerting rather
+	// than disable it. This issue can be found here
+	unifiedAlerting, err := section.Key("enabled").Bool()
 	if err != nil {
 		// TODO: Remove in Grafana v9
 		if cfg.IsFeatureToggleEnabled("ngalert") {
 			cfg.Logger.Warn("ngalert feature flag is deprecated: use unified alerting enabled setting instead")
-			enabled = true
-			// feature flag overrides the legacy alerting setting.
+			// feature flag overrides the legacy alerting setting
 			legacyAlerting := false
 			AlertingEnabled = &legacyAlerting
-			return &enabled, nil
+			unifiedAlerting = true
+			return &unifiedAlerting, nil
 		}
-		if IsEnterprise {
-			enabled = false
-			if AlertingEnabled == nil {
-				legacyEnabled := true
-				AlertingEnabled = &legacyEnabled
-			}
-			return &enabled, nil
-		}
-		// next, check whether legacy flag is set
-		if AlertingEnabled != nil && !*AlertingEnabled {
-			enabled = true
-			return &enabled, nil // if legacy alerting is explicitly disabled, enable the unified alerting by default.
+
+		// if legacy alerting has not been configured then enable unified alerting
+		if AlertingEnabled == nil {
+			unifiedAlerting = true
+			return &unifiedAlerting, nil
 		}
-		// NOTE: If the enabled flag is still not defined, the final decision is made during migration (see sqlstore.migrations.ualert.CheckUnifiedAlertingEnabledByDefault).
-		cfg.Logger.Info("The state of unified alerting is still not defined. The decision will be made during as we run the database migrations")
```

Comment:
Isn't this partially true? IIRC were introduced a safety check to avoid losing data so I _think_ there might be a case where a decision can be made at migration time - can you please double check?

Reply:
I understand that this was required as before this change we had the following:

| Alerting      | Unified Alerting | alerts table has alerts | Outcome |
| ------------- | ---------------- | ----------------------- | ------- |
| true          | ""               | true                    | Legacy  |
| true          | ""               | false                   | UA      |
| ""            | ""               | true                    | Legacy  |
| ""            | ""               | false                   | UA      |

but now we changed this table to:

| Alerting      | Unified Alerting | alerts table has alerts | Outcome |
| ------------- | ---------------- | ----------------------- | ------- |
| true          | ""               | true                    | UA      |
| true          | ""               | false                   | UA      |
| ""            | ""               | true                    | UA      |
| ""            | ""               | false                   | UA      |

i.e. whether or not the database has alerts is irrelevant to the outcome and there is no longer a decision to be made at migration time.